src/rustdoc-json-types/tests.rs

```diff
@@ -0,0 +1,34 @@
+use super::*;
+
+#[test]
+fn test_struct_info_roundtrip() {
```

Comment:
I can't run these. I think you need to do some things in `src/bootstrap` so it knows to run `cargo test` in ` src/rustdoc-json-types`. (Doing this my hand might have caused the `version` in `Cargo.toml`)

```
$ ./x.py test src/rustdoc-json-types
Updating only changed submodules
Submodules updated in 0.02 seconds
    Finished dev [unoptimized + debuginfo] target(s) in 2.26s
thread 'main' panicked at 'error: no rules matched src/rustdoc-json-types', src/bootstrap/builder.rs:243:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
failed to run: /home/nixon/upstreams/rust/rust/build/bootstrap/debug/bootstrap test src/rustdoc-json-types
Build completed unsuccessfully in 0:00:03
```

Reply:
I think you could add it as a `host_test!`: https://github.com/rust-lang/rust/blob/51748a8fc77283914d4135f31b5966a407208187/src/bootstrap/test.rs#L977

Reply:
I think running by hand changed it, yeah. I'll revert the version = 3 change, and fix up bootstrap.

Reply:
`x.py test src/rustdoc-json-types` now runs the tests as expected